src/libraries/Microsoft.Extensions.Configuration.Binder/gen/SourceGenerationSpec.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Configuration.Binder.SourceGeneration
+{
+    internal sealed record SourceGenerationSpec(
+        HashSet<TypeSpec> TypesForBindMethodGen,
```

Comment:
As with lists, `HashSet` uses reference equality so incremental compilation will not work as expected. Assuming you have a list with sequence equality (like the one I shared with you) then it should be possible to obtain set equality if you pre-sort the entries using a deterministic key (such as the FQN of the generated types).

Reply:
💡 This was my solution:

https://github.com/DotNetAnalyzers/StyleCopAnalyzers/commit/253d408d44f877cf8661c8bb1f84bb58b20cffa6

If there is a better solution I'm interested to know.